compiler/passes.nim

```diff
@@ -88,11 +92,15 @@ proc processImplicits(graph: ModuleGraph; implicits: seq[string], nodeKind: TNod
   for module in items(implicits):
     # implicit imports should not lead to a module importing itself
     if m.position != resolveMod(graph.config, module, relativeTo).int32:
-      var importStmt = newNodeI(nodeKind, m.info)
-      var str = newStrNode(nkStrLit, module)
-      str.info = m.info
-      importStmt.add str
-      if not processTopLevelStmt(graph, importStmt, a): break
+      var code = ""
+      let quoted = "".dup(addQuoted(module))
+      let name = module.splitFile.name
+      case nodeKind
+      of nkImportStmt: code = "import $1 as $2\n{.used: $2.}" % [quoted, name]
+      of nkIncludeStmt: code = "include $1" % quoted
+      else: assert false
+      let node = parseString(code, graph.cache, graph.config, filename = "", line = 0, errorHandler = nil)
```

Comment:
What? That's terrible. Why not use an AST construction macro instead?

Reply:
Why is this so terrible? the traditional way based on manually constructing the AST via `newNodeI` etc is arguably more complex to write and maintain; even with the above simple snippet (`"import $1 as $2\n{.used: $2.}"`) it's hard to write the corresponding code manually.

> Why not use an AST construction macro instead?

which one, or how would it work? if it doesn't exist yet i suggest to delay this to future work; this PR is already quite complex and solves many hard to fix long standing bugs (4 filed issues plus other issues I didn't bother writing bugs for).

Reply:
Well this style is simply not acceptable. Since it's a bugfix, why not fix the bug instead of pushing for idioms that are dubious at best. (How do we know `"".dup(addQuoted(module))` respects Nim's quoting rules?)

Reply:
> Why not use an AST construction macro instead?

done